lib/govuk_index/document_type_inferer.rb

```diff
@@ -21,10 +23,20 @@ def unpublishing_type?
 
   private
 
-    attr_reader :payload
+    attr_reader :payload, :payload_document_type
+
+    def mapped_document_types
+      @_document_types ||= begin
+        YAML.load_file(File.join(__dir__, '../../config/govuk_index/mapped_document_types.yaml'))
+      end
+    end
 
     def existing_document
       @_existing_document ||= Client.get(type: '_all', id: payload['base_path'])
     end
+
+    def elasticsearch_document_type
+      @_elasticsearch_document_type ||= mapped_document_types[payload_document_type]
```

Comment:
`payload_document_type` is only used the once so may be worthwhile just inlining `payload['document_type']`